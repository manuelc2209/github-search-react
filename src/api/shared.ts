export const BASE_PATH = "https://api.github.com";

export interface Data {
  items: UserProps[];
}
export interface UserProps {
  avatar_url?: string;
  events_url?: string;
  followers_url?: string;
  following_url?: string;
  gists_url?: string;
  gravatar_id?: string;
  html_url?: string;
  id?: number;
  login?: string;
  rateLimited?: boolean;
  node_id?: string;
  organizations_url?: string;
  received_events_url?: string;
  repos_url?: string;
  score?: number;
  site_admin?: boolean;
  starred_url?: string;
  subscriptions_url?: string;
  type?: string;
  url?: string;
  profileData?: any;
}
