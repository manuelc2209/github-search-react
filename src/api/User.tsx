import { BASE_PATH, Data } from "./shared";

let githubClientId = process.env.REACT_APP_GITHUB_CLIENT_ID;
let githubClientSecret = process.env.REACT_APP_GITHUB_CLIENT_SECRET;

export interface userData {}

export async function getUserData(user?: string, url?: string) {
  try {
    const res = await fetch(
      url
        ? `${url}?client_id=${githubClientId}&client_secret=${githubClientSecret}`
        : `${BASE_PATH}/search/users?q=${user}&client_id=${githubClientId}&client_secret=${githubClientSecret}`
    );

    if (!res.ok) {
      const message = `An error has occured: ${res.status} - ${res.statusText}`;
      throw new Error(message);
    }

    const data = await res.json();

    const result = {
      status: res.status,
      headers: {
        "Content-Type": res.headers.get("Content-Type"),
        "Content-Length": res.headers.get("Content-Length"),
      },
      length: res.headers.get("Content-Length"),
      data: data,
    };
    return result;
  } catch (err: any) {
    return err.message;
  }
}

export async function buildUserProfile(data: Data) {
  const newData = data.items.map((item: any) =>
    getUserData(undefined, item.url).then((profileData) => {
      return {
        ...item,
        profileData: profileData.data,
      };
    })
  );

  return await Promise.all(newData).then(function (results) {
    return results;
  });
}
