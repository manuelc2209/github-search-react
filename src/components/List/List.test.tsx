import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { List } from "./";
import { DataProps } from "../../api/shared";

const userData = {
  login: "ads",
  id: 131052,
  node_id: "MDQ6VXNlcjEzMTA1Mg==",
  avatar_url: "https://avatars.githubusercontent.com/u/131052?v=4",
  gravatar_id: "",
  url: "https://api.github.com/users/ads",
  html_url: "https://github.com/ads",
  followers_url: "https://api.github.com/users/ads/followers",
  following_url: "https://api.github.com/users/ads/following{/other_user}",
  gists_url: "https://api.github.com/users/ads/gists{/gist_id}",
  starred_url: "https://api.github.com/users/ads/starred{/owner}{/repo}",
  subscriptions_url: "https://api.github.com/users/ads/subscriptions",
  organizations_url: "https://api.github.com/users/ads/orgs",
  repos_url: "https://api.github.com/users/ads/repos",
  events_url: "https://api.github.com/users/ads/events{/privacy}",
  received_events_url: "https://api.github.com/users/ads/received_events",
  type: "User",
  site_admin: false,
  profileData: {
    name: "asd",
    id: 123123123123,
    company: "home",
    email: "a@email.com",
    username: "asdUsername",
    avatar_url: "http://www.google.com",
  },
  score: 1,
};

const orgData = {
  ...userData,
  type: "Organization",
};

const pushNumber = 5;
const organizations: Array<DataProps> = [];
const users: Array<DataProps> = [];
for (let i = 0; i < pushNumber; i++) {
  organizations.push(orgData);
  users.push(userData);
}

test("render a List with dummy data", () => {
  const component = render(<List data={users} />);
  expect(component).toMatchSnapshot();
});

test("render a List an empty data", () => {
  const component = render(<List data={[]} />);
  expect(component).toMatchSnapshot();
});

test("render a List and click on Show More", async () => {
  const component = render(<List data={organizations} />);
  const button = await component.findByTestId("StyledButton");
  button.focus();
  fireEvent.click(button);
  expect(component).toMatchSnapshot();
});
