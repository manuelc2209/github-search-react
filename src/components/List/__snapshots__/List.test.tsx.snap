// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a List an empty data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render a List and click on Show More 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fFeiMQ eTnIod"
      >
        <div
          class="sc-hGPBjI byVhgN"
        >
          <div
            class="sc-bkkeKt cUEpkn"
          >
            <span
              class="sc-ieecCq dZJVmw"
            >
              Results(5)
            </span>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    Organization
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    Organization
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    Organization
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    Organization
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    Organization
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fFeiMQ eTnIod"
    >
      <div
        class="sc-hGPBjI byVhgN"
      >
        <div
          class="sc-bkkeKt cUEpkn"
        >
          <span
            class="sc-ieecCq dZJVmw"
          >
            Results(5)
          </span>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  Organization
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  Organization
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  Organization
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  Organization
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  Organization
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`render a List with dummy data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fFeiMQ eTnIod"
      >
        <div
          class="sc-hGPBjI byVhgN"
        >
          <div
            class="sc-bkkeKt cUEpkn"
          >
            <span
              class="sc-ieecCq dZJVmw"
            >
              Results(5)
            </span>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    User
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    User
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    User
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
              >
                <div
                  class="sc-dkPtRN cUvaGP"
                >
                  <img
                    class="sc-bdvvtL ionVky"
                    src="https://avatars.githubusercontent.com/u/131052?v=4"
                  />
                  <div
                    class="sc-jRQBWg iANpjD"
                  >
                    <h5
                      class="sc-hKwDye hWhLRG"
                    >
                      Name: 
                      asd
                    </h5>
                    <h6
                      class="sc-eCImPb hcWkvY"
                    >
                      User Email: 
                      a@email.com
                    </h6>
                    <h6>
                      Username: 
                      ads
                    </h6>
                    <p
                      class="sc-gKclnd gefYSt"
                    >
                      Company: 
                      home
                    </p>
                  </div>
                  <span
                    class="sc-iCfMLu iGnodU"
                  >
                    User
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="sc-dJjYzT fnXtUE"
          data-testid="StyledButton"
        >
          Click to show more
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fFeiMQ eTnIod"
    >
      <div
        class="sc-hGPBjI byVhgN"
      >
        <div
          class="sc-bkkeKt cUEpkn"
        >
          <span
            class="sc-ieecCq dZJVmw"
          >
            Results(5)
          </span>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  User
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  User
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  User
                </span>
              </div>
            </div>
          </div>
          <div>
            <div
              class="sc-gsDKAQ cFOpLH sc-fotOHu clLcQV"
            >
              <div
                class="sc-dkPtRN cUvaGP"
              >
                <img
                  class="sc-bdvvtL ionVky"
                  src="https://avatars.githubusercontent.com/u/131052?v=4"
                />
                <div
                  class="sc-jRQBWg iANpjD"
                >
                  <h5
                    class="sc-hKwDye hWhLRG"
                  >
                    Name: 
                    asd
                  </h5>
                  <h6
                    class="sc-eCImPb hcWkvY"
                  >
                    User Email: 
                    a@email.com
                  </h6>
                  <h6>
                    Username: 
                    ads
                  </h6>
                  <p
                    class="sc-gKclnd gefYSt"
                  >
                    Company: 
                    home
                  </p>
                </div>
                <span
                  class="sc-iCfMLu iGnodU"
                >
                  User
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="sc-dJjYzT fnXtUE"
        data-testid="StyledButton"
      >
        Click to show more
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
